src/mlpack/tests/mean_shift_test.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file mean_shift_test.cpp
+ * @author Shangtong Zhang
+ */
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/mean_shift/mean_shift.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace mlpack::meanshift;
+
+BOOST_AUTO_TEST_SUITE(MeanShiftTest);
+
+// Generate dataset; written transposed because it's easier to read.
+arma::mat meanShiftData("  0.0   0.0;" // Class 1.
+                     "  0.3   0.4;"
+                     "  0.1   0.0;"
+                     "  0.1   0.3;"
+                     " -0.2  -0.2;"
+                     " -0.1   0.3;"
+                     " -0.4   0.1;"
+                     "  0.2  -0.1;"
+                     "  0.3   0.0;"
+                     " -0.3  -0.3;"
+                     "  0.1  -0.1;"
+                     "  0.2  -0.3;"
+                     " -0.3   0.2;"
+                     " 10.0  10.0;" // Class 2.
+                     " 10.1   9.9;"
+                     "  9.9  10.0;"
+                     " 10.2   9.7;"
+                     " 10.2   9.8;"
+                     "  9.7  10.3;"
+                     "  9.9  10.1;"
+                     "-10.0   5.0;" // Class 3.
+                     " -9.8   5.1;"
+                     " -9.9   4.9;"
+                     "-10.0   4.9;"
+                     "-10.2   5.2;"
+                     "-10.1   5.1;"
+                     "-10.3   5.3;"
+                     "-10.0   4.8;"
+                     " -9.6   5.0;"
+                     " -9.8   5.1;");
+
+
+/**
+ * 30-point 3-class test case for Mean Shift.
+ */
+BOOST_AUTO_TEST_CASE(MeanShiftSimpleTest) {
+ 
+  MeanShift<> meanShift;
+  
+  arma::Col<size_t> assignments;
+  arma::mat centroids;
+  meanShift.Cluster((arma::mat) trans(meanShiftData), assignments, centroids);
+  
+  // Now make sure we got it all right.  There is no restriction on how the
+  // clusters are ordered, so we have to be careful about that.
+  size_t firstClass = assignments(0);
+  
+  for (size_t i = 1; i < 13; i++)
+    BOOST_REQUIRE_EQUAL(assignments(i), firstClass);
+  
+  size_t secondClass = assignments(13);
+  
+  // To ensure that class 1 != class 2.
+  BOOST_REQUIRE_NE(firstClass, secondClass);
+  
+  for (size_t i = 13; i < 20; i++)
+    BOOST_REQUIRE_EQUAL(assignments(i), secondClass);
+  
+  size_t thirdClass = assignments(20);
+  
+  // To ensure that this is the third class which we haven't seen yet.
+  BOOST_REQUIRE_NE(firstClass, thirdClass);
+  BOOST_REQUIRE_NE(secondClass, thirdClass);
+  
+  for (size_t i = 20; i < 30; i++)
+    BOOST_REQUIRE_EQUAL(assignments(i), thirdClass);
+  
+}
+
```

Comment:
Once we merge this I will probably add at least one more test where I generate observations from a mixture of Gaussians (that are sufficiently far apart) and ensure that the mean shift algorithm will properly label each point.  You can do this now if you like, but I wanted to let you know that I'm planning to do that.  More tests, in general, is a good thing.  It may be worth adding some simple tests like "if I set the duplicate threshold to 0, do I get dataset.n_cols clusters?"; this is helpful to people who might be modifying the code later on and want to make sure it still functions the same way.